tests.py

```diff
@@ -0,0 +1,81 @@
+from battery import Battery
+from limit_checker import LimitChecker
+class Test:
+    def __init__(self) -> None:
+        self.test_config = [{"Temperature": 50, "SOC": 85, "ChargeRate": 0, "Result": False},
+        {"Temperature": 25, "SOC": 70, "ChargeRate": 0.7, "Result": True}]
+
+        self.temperature_config = [{"Temperature":100, "Result": False},
+         {"Temperature": -20, "Result": False},
+         {"Temperature": 34, "Result": True}]
+        
+        self.soc_config = [{"SOC":120, "Result": False},
+         {"SOC": -2, "Result": False},
+         {"SOC": 25, "Result": True}]
+
+        self.charge_rate_config = [{"ChargeRate":10, "Result": False},
+         {"ChargeRate": 1, "Result": False},
+         {"ChargeRate": 0.1, "Result": True}]
+
+    def test_battery(self):
+        for config in self.test_config:
+            battery = Battery()
+            battery.temperature = config.get("Temperature")
+            battery.soc = config.get("SOC")
+            battery.charge_rate = config.get("ChargeRate")
+            battery.is_battery_ok()
```

Comment:
is_battery_ok tests can be extended to cover many other values. Please see if that can be done.